### PR TITLE
Add no workspace UI for giskard

### LINKED ISSUE #1343: No workspace UI

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
The no workspace error handling dialog.

### Relationships
Closes #1343
Sibling of #1342

## Why do we want this?
We get into this nasty state sometimes, we can't let our users get blocked by it.

## How is it done?
This is a basic MvxDialogFragment with basic bindings. Code is straightforward. The vm and error handling were already there for iOS.

### Why not another way?
We are tied to the mvx navigation scheme and it works well for this case. It's rather simple. Let me know if you have a better idea.

### Side effects
0️⃣

## Review hints
Everything is new, I tried making review by commit easy but it doesn't make much of a difference here.
Instructions on how to get into NoWorkspace state can be found here: https://github.com/toggl/mobileapp/issues/2782#issuecomment-410079411

## :squid: Permissions
🦑 whenever

### CODE REVIEW

**Comment by kornelijepetak** [comment]
Ok, the app crashes for me at random times. 
First time it just crashed when I logged in.
Next time, after login, I could see the no-workspace screen for a split second and then it went to the main log screen.
Third time, after login, it crashed. But when I reopened the app from the app drawer, it showed the no-workspace screen correctly.

Can't consistently reproduce. :/

**Comment by daividssilverio** [comment]
Indeed. I got to reproduce it too :/ Exact same 3 occurrences. Crash, split second, working.
I have no clue... I'll investigate.

**Comment by daividssilverio** [comment]
Ok... we have a nasty race condition... The MainViewModel is appearing after the NoWorkspaceViewModel on startup :/

**Comment by daividssilverio** [comment]
Fixed @kornelijepetak o/ Everything should work just fine now.

It would be really great now if we could have a functional test for iOS as well.

File: Toggl.Foundation.MvvmCross/ViewModels/NoWorkspaceViewModel.cs [hunk 1]
```diff
@@ -37,15 +46,17 @@ public async Task TryAgain()
...
             dataSource.CreateNewSyncManager();
 
-            var workspaces = await dataSource
+            var anyWorkspaceIsAvailable = await dataSource
                 .SyncManager
                 .ForceFullSync()
                 .Where(state => state == SyncState.Sleep)
-                .SelectMany(dataSource.Workspaces.GetAll());
+                .SelectMany(dataSource.Workspaces.GetAll())
+                .Where(workspaces => workspaces.Any())
```
Comment (kornelijepetak) [inline Toggl.Foundation.MvvmCross/ViewModels/NoWorkspaceViewModel.cs]: You can merge `Where` and `Any` into
`.Any(workspaces => workspaces.Any())`

File: Toggl.Giskard/Fragments/NoWorkspaceFragment.cs [hunk 2]
```diff
@@ -29,14 +31,18 @@ public NoWorkspaceFragment(IntPtr javaReference, JniHandleOwnership transfer)
...
+            base.OnCreateView(LayoutInflater, container, savedInstanceState);
             var rootView = inflater.Inflate(Resource.Layout.NoWorkspaceFragment, container, false);
             InitializeViews(rootView);
             this.Bind(createWorkspaceTextView.Tapped(), ViewModel.CreateWorkspaceWithDefaultName);
             this.Bind(tryAgainTextView.Tapped(), ViewModel.TryAgain);
-
             this.Bind(ViewModel.IsLoading.Select(CommonFunctions.Invert), createWorkspaceTextView.BindEnabled());
             this.Bind(ViewModel.IsLoading.Select(CommonFunctions.Invert), tryAgainTextView.BindEnabled());
-            this.Bind(ViewModel.IsLoading.StartWith(false), progressBar.BindIsVisible());
+            this.Bind(ViewModel.IsLoading.StartWith(false), isLoading =>
```
Comment (kornelijepetak) [inline Toggl.Giskard/Fragments/NoWorkspaceFragment.cs]: please, name this function. Otherwise bindings will soon become unbecoming.

**Comment by daividssilverio** [comment]
Try some more @kornelijepetak, please o/

**Pull request merged**